front-end-tools/src/components/InitComponent.tsx

```diff
@@ -0,0 +1,641 @@
+
+                            setUploadedModuleSchemaBase64(undefined);
+
+                            const checkboxElement = form.getValues('useModuleReferenceFromStep1');
+
+                            form.setValue('moduleReference', undefined);
+
+                            if (
+                                checkboxElement &&
+                                moduleReferenceDeployed === undefined &&
+                                moduleReferenceCalculated === undefined
+                            ) {
+                                setModuleReferenceError('Module reference is not set in step 1');
+                            }
+
+                            const newModuleReference =
+                                moduleReferenceDeployed !== undefined
+                                    ? moduleReferenceDeployed
+                                    : moduleReferenceCalculated;
+
+                            if (checkboxElement && newModuleReference !== undefined) {
+                                form.setValue('moduleReference', newModuleReference);
+
+                                setDisplayContracts(contracts);
+                                form.setValue('smartContractName', contracts[0]);
+                            }
+                        }}
+                    />
+                </Form.Group>
+
+                {useModuleReferenceFromStep1 && (
+                    <>
+                        <br />
+                        <Alert variant="info">
+                            <div>
+                                This checkbox autofilled the <code>module reference</code>, the{' '}
+                                <code>smart contract name</code>, and the <code>input parameter schema</code> from the
+                                module in step1.
+                            </div>
+                            <br />
+                            <div>
+                                <b>Uncheck</b> this box, if you want to manually fill in a <code>module reference</code>
+                                , the <code>smart contract name</code>, or an <code>input parameter schema</code>.
+                            </div>
+                            <br />
+                            <div>
+                                <b>Uncheck</b> and <b>check</b> this box again, if you want to load a new module from
+                                step 1.
+                            </div>
+                        </Alert>
+                        <br />
+                    </>
+                )}
+
+                {moduleReferenceError && <Alert variant="danger"> Error: {moduleReferenceError}. </Alert>}
+
+                <Row>
+                    <Form.Group className="col-md-4 mb-3">
+                        <Form.Label> Module Reference</Form.Label>
+                        <Form.Control
+                            defaultValue="91225f9538ac2903466cc4ab07b6eb607a2cd349549f357dfdf4e6042dde0693"
```

Comment:
hmm, that would suggest that whenever we upload a new module in step 1 we automatically update values in box 2 (such as this `defaultValue` as well as its underlying field value). I am a bit worried that this will get confusing at some point if values across boxes change so I tried to avoid it.

Reply:
It's fine to not do this now, but it's not the best experience as it is now either.

Perhaps this will be cleaner when/if the pages are split into the different flows, so that not everything is on the same page.